lib/repositories/providers/agenda_provider.dart

```diff
@@ -1,18 +1,52 @@
-
-
 import 'package:dio/dio.dart';
 import 'package:dio_http_cache/dio_http_cache.dart';
+import 'package:intl/intl.dart';
 import 'package:v34/models/event.dart';
 import 'package:v34/repositories/providers/http.dart';
 
 class AgendaProvider {
+
+  static DateTime roundDownToHour(DateTime dt) {
+    return dt.subtract(
+      Duration(
+        microseconds: dt.microsecond,
+        milliseconds: dt.millisecond,
+        seconds: dt.second,
+        minutes: dt.minute
+      )
+    );
+  }
   
   Future<List<Event>> listEvents() async {
     Response response = await dio.get("/calendars", options: buildConfigurableCacheOptions());
     if (response.statusCode == 200) {
       return (response.data as List).map((json) => Event.fromJson(json)).toList();
     } else {
-      throw Exception('Impossible de récupérer les clubs');
+      throw Exception('Impossible de récupérer les événements');
+    }
+  }
+
+  Future<List<Event>> listTeamMonthMatches(String teamCode) async {
+    Response response = await dio.get("/equipes/$teamCode/matchs", options: buildConfigurableCacheOptions());
+    if (response.statusCode == 200) {
+      List<Event> events = (response.data as List).map((json) => Event.fromJson(json)).toList();
+      events.removeWhere((event) => event.date.compareTo(DateTime.now()) < 0 || event.date.compareTo(DateTime.now().add(Duration(days: 30))) > 0);
+      return events;
+    } else {
+      throw Exception('Impossible de récupérer les événements');
+    }
+  }
+
+  Future<List<Event>> listTeamMonthEvents(String teamCode) async {
+    DateTime min = roundDownToHour(DateTime.now());
+    DateTime max = min.add(Duration(days: 30));
+    DateFormat format = DateFormat("yyyy-MM-ddTHH:mm:ss");
```

Comment:
Instead of `roundDownToHour` min and max, can't you just change the `DateFormat` pattern and fix minutes and seconds? Something like `DateFormat("yyyy-MM-ddTHH:00:00")` 
And even if you can't, what's the impact of using minutes & seconds?

Reply:
Yes it's smart !

Reply:
I round to optimize regarding to the cache. If we take the exact time, each request will not be the same and then we don't use the cache.